tests/test/org/commcare/util/test/SqlStorageIndexedTests.java

```diff
@@ -0,0 +1,156 @@
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.Vector;
+
+import static junit.framework.Assert.fail;
+import static junit.framework.TestCase.assertFalse;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class SqlStorageIndexedTests {
+
+    Case a, b, c, d, e;
+
+    Ledger l, l2, l3;
+
+    SqlIndexedStorageUtility<Case> caseStorage;
+    SqlIndexedStorageUtility<Ledger> ledgerStorage;
+    String owner;
+    String groupOwner;
+    String otherOwner;
+    Vector<String> groupOwned;
+    Vector<String> userOwned;
+
+
+    @Before
+    public void setUp() throws Exception {
+
+        owner = "owner";
+        otherOwner = "otherowner";
+        groupOwner = "groupowned";
+
+        userOwned = new Vector<String>();
+        userOwned.addElement(owner);
+
+        groupOwned = new Vector<String>();
+        groupOwned.addElement(owner);
+        groupOwned.addElement(groupOwner);
+
+        a = new Case("case_name_ipsum", "case_type_ipsum");
+        a.setCaseId("case_id_ipsum");
+        a.setUserId(owner);
+        a.setID(-1);
+
+        l = new Ledger("ledger_entity_id");
+        l.setID(-1);
+        l.setEntry("test_section_id", "test_entry_id", 2345);
+
+        l2 = new Ledger("ledger_entity_id");
+        l2.setID(-1);
+        l2.setEntry("test_section_id_2", "test_entry_id_2", 2345);
+
+        l3 = new Ledger("ledger_entity_id_3");
+        l3.setID(-1);
+        l3.setEntry("test_section_id_3", "test_entry_id_3", 2345);
+    }
+
+    @Test
+    public void testSqlCaseStorage() {
+        try {
+
+            try {
+
+                PrototypeFactory mPrototypeFactory = new PrototypeFactory();
+                mPrototypeFactory.addClass(Case.class);
+
+                String storageKey = "TFCase";
+                String username = "sql-storage-test";
+
+                SqlTestUtils.deleteDatabase(username);
+
+                caseStorage = new SqlIndexedStorageUtility<Case>(Case.class, username, storageKey);
+
+                caseStorage.write(a);
+
+                Case readCase = caseStorage.read(1);
+
+                assertEquals("case_name_ipsum", readCase.getName());
+
+            } catch ( Exception e ) {
+                System.err.println(e.getClass().getName() + ": " + e.getMessage());
+                e.printStackTrace();
+                System.exit(0);
+            }
+
+        } catch (Exception e) {
+            e.printStackTrace();
+            fail("Unexpected exception " + e.getMessage());
+        }
+    }
+
+    @Test
+    public void testSqlLedgerStorage() {
+        try {
+
+            try {
+
+                PrototypeFactory mPrototypeFactory = new PrototypeFactory();
+                mPrototypeFactory.addClass(Ledger.class);
+
+                String storageKey = "Ledger";
+                String username = "wspride";
+
+                SqlTestUtils.deleteDatabase(username);
```

Comment:
I would argue that removing old databases and the values in them should happen at the end of tests that created those values, not at the beginning of other tests.
Additionally, would be nice to make sure DB related tests remove the `*.db` files created when they are done so we don't have files floating around our commcare directories